classes/Wrapper/ProductWrapper.php

```diff
@@ -20,97 +20,237 @@
+            'item_id' => (int) $product_id,
+            'item_name' => (string) $product['name'],
+            'affiliation' => Shop::isFeatureActive() ? $this->context->shop->name : Configuration::get('PS_SHOP_NAME'),
+            'index' => 0,
+            'price' => $product['price_amount'],
+            'quantity' => 1,
         ];
+
+        // Add manufacturer info if we have it
+        if (!empty($product['manufacturer_name'])) {
+            $item['item_brand'] = $product['manufacturer_name'];
+        // If we don't, which can happen due to some bugs in getProductProperties, we will fetch it manually
+        } elseif (!empty($product['id_manufacturer'])) {
+            $manufacturerName = Manufacturer::getNameById((int) $product['id_manufacturer']);
+            if (!empty($manufacturerName)) {
+                $item['item_brand'] = $manufacturerName;
+            }
+        }
+
+        // We will specify variant ID if we have it
+        if (!empty($product['id_product_attribute'])) {
+            $item['item_id'] .= '-' . $product['id_product_attribute'];
+        }
+
+        // Information about a chosen variant, if we have it (cart list has this out of the box)
+        if (!empty($product['attributes_small'])) {
+            $item['item_variant'] = $product['attributes_small'];
+
+        // If we don't, we will construct it in the same format
+        } elseif (!empty($product['id_product_attribute'])) {
+            $variant = $this->getProductVariant((int) $product['id_product_attribute']);
+            if (!empty($variant)) {
+                $item['item_variant'] = $variant;
+            }
+        }
+
+        if ($useProvidedQuantity === true) {
+            // Info about quantity in cart, if we have it
+            $item['quantity'] = $product['quantity'];
+        }
+
+        // Prepare category information, if not loaded before, we will fetch it
+        if (!isset($this->categories[(int) $product_id])) {
+            $this->loadCategories([(int) $product_id]);
+        }
+
+        $productCategories = $this->categories[(int) $product_id];
+
+        // If the category is our default one, we will move it to the beginning of that list
+        foreach ($productCategories as $key => $category) {
+            if ($category['id_category'] == $product['id_category_default']) {
+                $productCategories = [$key => $category] + $productCategories;
+                break;
+            }
+        }
+
+        // Add it to our item
+        $counter = 1;
+        foreach ($productCategories as $productCategory) {
+            // Skip home category if we have more than 1 category
+            if (count($productCategories) > 1 && $productCategory['id_category'] == $this->homeCategory) {
+                continue;
+            }
+
+            // Add it with proper key
+            $item[$counter == 1 ? 'item_category' : 'item_category' . $counter] = $productCategory['name'];
+            if ($counter == 5) {
```

Comment:
Again some rules 😄 that are hard to understand. What is special with counter 5?

Sorry for asking all these questions, but I'm sure the next developer who tries to read the code will ask the same 😉

Reply:
https://developers.google.com/analytics/devguides/collection/ga4/reference/events?client_type=gtag#view_item

Because we can have maximum of item_category5 in the event.